tasks/balances.ts

```diff
@@ -27,69 +39,122 @@ const balancesError = `
+      networkName: "btc_testnet",
+    };
+  } catch (error) {}
+};
+
+const fetchNativeBalance = async (address: string, provider: any) => {
   const balance = await provider.getBalance(address);
-  return parseFloat(ethers.utils.formatEther(balance)).toFixed(2);
+  return parseFloat(hre.ethers.utils.formatEther(balance)).toFixed(2);
 };
 
 const fetchZetaBalance = async (
   address: string,
-  provider: ethers.providers.JsonRpcProvider,
+  provider: any,
   networkName: string
 ) => {
   if (networkName === "athens") return "";
-  const zetaAddress = getAddress({
-    address: "zetaToken",
-    networkName,
-    zetaNetwork: "athens",
-  });
-  const contract = new ethers.Contract(zetaAddress, ZetaEth, provider);
+  const zetaAddress = getAddress("zetaToken", networkName as any);
+  const contract = new hre.ethers.Contract(zetaAddress, ZetaEth.abi, provider);
   const balance = await contract.balanceOf(address);
-  return parseFloat(ethers.utils.formatEther(balance)).toFixed(2);
+  return parseFloat(hre.ethers.utils.formatEther(balance)).toFixed(2);
 };
 
-const main = async (args: any, hre: HardhatRuntimeEnvironment) => {
-  const { ethers, config } = hre as any;
-  const fetchBalances = async (
-    address: string,
-    provider: ethers.providers.JsonRpcProvider,
-    networkName: string
-  ) => {
-    try {
-      const { config } = hre as any;
-      const { url } = config.networks[networkName];
-      const native = await fetchNativeBalance(address, provider);
-      const zeta = await fetchZetaBalance(address, provider, networkName);
+const fetchBalances = async (
+  address: string,
+  provider: any,
+  networkName: string
+) => {
+  try {
+    const native = await fetchNativeBalance(address, provider);
+    const zeta = await fetchZetaBalance(address, provider, networkName);
+    const isZeta = networkName === "zeta_testnet";
+    const zrc20 = isZeta ? { zrc20: await fetchZRC20Balance(address) } : {};
+    return { native, networkName, zeta, ...zrc20 };
+  } catch (error) {}
+};
 
-      return { native, networkName, zeta };
+const fetchZRC20Balance = async (address: string) => {
+  const { url } = hre.config.networks["zeta_testnet"] as any;
+  const provider = new hre.ethers.providers.JsonRpcProvider(url);
+
+  const promises = Object.keys(hre.config.networks).map(async (networkName) => {
+    try {
+      const zrc20 = getAddress("zrc20", networkName);
+      const contract = new hre.ethers.Contract(zrc20, ZRC20.abi, provider);
+      const balance = await contract.balanceOf(address);
+      const denom = networks[networkName].assets[0].symbol;
+      if (balance > 0) {
+        const b = parseFloat(hre.ethers.utils.formatEther(balance)).toFixed(2);
+        return `${b} ${denom}`;
+      }
     } catch (error) {}
-  };
+  });
+
+  const result = await Promise.all(promises);
+
+  // tBTC ZRC-20 balance
+  const btcZRC20 = "0x65a45c57636f9BcCeD4fe193A602008578BcA90b"; // use getAddress("zrc20", "btc_testnet") when available
```

Comment:
maybe any comments that need actions could be marked with `TODO:`?

Reply:
https://github.com/zeta-chain/toolkit/pull/10/commits/e9324e92357b1657d8cb8b47bda47ac6215bb45f